packages/core/src/decorators/UseRequestContext.ts

```diff
@@ -1,15 +1,15 @@
 import { MikroORM } from '../MikroORM';
 import { RequestContext } from '../utils/RequestContext';
 
-export function UseRequestContext(getContext?: MikroORM | (() => MikroORM)) {
+export function UseRequestContext<T>(getContext?: MikroORM | ((type?: T) => MikroORM)): MethodDecorator {
   return function (target: any, propertyKey: string | symbol, descriptor: PropertyDescriptor) {
     const originalMethod = descriptor.value;
-    descriptor.value = async function (this: { orm: MikroORM }, ...args: any[]) {
+    descriptor.value = async function (this: T, ...args: any[]) {
       /* istanbul ignore next */
-      const orm = getContext instanceof MikroORM ? getContext : (getContext?.() ?? this.orm);
+      const orm = getContext instanceof MikroORM ? getContext : (getContext?.(this) ?? (this as any).orm);
 
       if (!(orm as unknown instanceof MikroORM)) {
-        throw new Error('@UseRequestContext() decorator can only be applied to methods of classes with `orm: MikroORM` property, or with a callback parameter like `@UseRequestContext(() => orm)`');
+        throw new Error('@UseRequestContext() decorator can only be applied to methods of classes with `orm: MikroORM` property, or with a callback parameter like `@UseRequestContext(type => type.orm)`');
```

Comment:
i would probably keep the old error message. this new use case is rather edge case, and the error now suggest it as it would be the go to way. it is not, the parameter signature is there to support providing the orm from outside, not from the class level.

```suggestion
        throw new Error('@UseRequestContext() decorator can only be applied to methods of classes with `orm: MikroORM` property, or with a callback parameter like `@UseRequestContext(() => orm)`');
```

or we would need to rephrase it more